lib/src/kvs/bootstrap/archive.rs

```diff
@@ -0,0 +1,528 @@
+use std::time::Duration;
+
+use rand::Rng;
+use tokio::sync::mpsc;
+use tokio::sync::oneshot;
+
+use crate::err::BootstrapCause::{ChannelRecvError, ChannelSendError};
+use crate::err::ChannelVariant::{BootstrapArchive, BootstrapTxSupplier};
+use crate::err::Error;
+use crate::kvs::bootstrap::{TxRequestOneshot, TxResponseOneshot};
+use crate::kvs::{ds, BootstrapOperationResult};
+use crate::sql::Uuid;
+
+/// This task will read input live queries from a receiver in batches and
+/// archive them and finally send them to the output channel.
+/// The task terminates if there is an irrecoverable error or if the input
+/// channel has been closed (dropped, from previous task).
+pub(crate) async fn archive_live_queries(
+	tx_req: mpsc::Sender<TxRequestOneshot>,
+	node_id: Uuid,
+	mut scan_recv: mpsc::Receiver<BootstrapOperationResult>,
+	sender: mpsc::Sender<BootstrapOperationResult>,
+	batch_size: usize,
+	batch_latency: &Duration,
+) -> Result<(), Error> {
+	let mut msg: Vec<BootstrapOperationResult> = Vec::with_capacity(batch_size);
+	loop {
+		match tokio::time::timeout(*batch_latency, scan_recv.recv()).await {
+			Ok(Some(bor)) => {
+				let is_err = bor.1.is_some();
+				if is_err {
+					// send any errors further on, because we don't need to process them
+					// unless we can handle them. Currently we can't.
+					// if we error on send, then we bubble up because this shouldn't happen
```

Comment:
This means the receiver is closed or dropped. I think we should simply log the fact and `break` here, not bubble the error up. Callers can't do anything to handle this.

Reply:
This is actually not about the channel, but the input message from the channel. This is under the `Some(bootstrap_message)` branch, so the channel isn't closed. But the input message is an error.

The reason for such a convoluted way of sending messages is because
- maybe we can handle the type of error
- maybe we still need to do something with the key
- we probably don't want errors preventing cluster nodes from being cleared, something that I was considering for that design

Reply:
Renamed variable